management/server/http/events_handler.go

```diff
@@ -45,22 +45,50 @@ func (h *EventsHandler) GetAllEvents(w http.ResponseWriter, r *http.Request) {
 		util.WriteError(err, w)
 		return
 	}
-	events := make([]*api.Event, 0)
+	events := make([]*api.Event, 0, len(accountEvents))
 	for _, e := range accountEvents {
 		events = append(events, toEventResponse(e))
 	}
 
+	err = h.fillEventsWithInitiatorEmail(events, account.Id, user.Id)
+	if err != nil {
+		util.WriteError(err, w)
+		return
+	}
+
 	util.WriteJSONObject(w, events)
 }
 
+func (h *EventsHandler) fillEventsWithInitiatorEmail(events []*api.Event, accountId, userId string) error {
+	// build email map based on users
+	userInfos, err := h.accountManager.GetUsersFromAccount(accountId, userId)
+	if err != nil {
+		log.Errorf("failed to get users from account: %s", err)
+		return err
+	}
+
+	emails := make(map[string]string)
+	for _, ui := range userInfos {
+		emails[ui.ID] = ui.Email
+	}
+
+	// fill event with email of initiator
+	for _, event := range events {
+		if event.InitiatorEmail == "" {
+			event.InitiatorEmail = emails[event.InitiatorId]
```

Comment:
Well, if everything goes well it should not happen. It could happen only that case if the database is corrupted. I do not think so we should handle this error on this layer but I will consider.